src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -1,10 +1,15 @@
 *REMOVED*Microsoft.CodeAnalysis.TextDocument.Project.set -> void
 *REMOVED*Microsoft.CodeAnalysis.TextDocument.TextDocument() -> void
 Microsoft.CodeAnalysis.Options.DocumentOptionSet.WithChangedOption<T>(Microsoft.CodeAnalysis.Options.PerLanguageOption<T> option, T value) -> Microsoft.CodeAnalysis.Options.DocumentOptionSet
+Microsoft.CodeAnalysis.Rename.Renamer.RenameDocumentResult
+Microsoft.CodeAnalysis.Rename.Renamer.RenameDocumentResult.RenameSymbolSucceeded.get -> bool
+Microsoft.CodeAnalysis.Rename.Renamer.RenameDocumentResult.Solution.get -> Microsoft.CodeAnalysis.Solution
 Microsoft.CodeAnalysis.Project.RemoveDocuments(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.DocumentId> documentIds) -> Microsoft.CodeAnalysis.Project
 Microsoft.CodeAnalysis.Solution.RemoveAdditionalDocuments(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.DocumentId> documentIds) -> Microsoft.CodeAnalysis.Solution
 Microsoft.CodeAnalysis.Solution.RemoveAnalyzerConfigDocuments(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.DocumentId> documentIds) -> Microsoft.CodeAnalysis.Solution
 Microsoft.CodeAnalysis.Solution.RemoveDocuments(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.DocumentId> documentIds) -> Microsoft.CodeAnalysis.Solution
 Microsoft.CodeAnalysis.Solution.WithOptions(Microsoft.CodeAnalysis.Options.OptionSet options) -> Microsoft.CodeAnalysis.Solution
+static Microsoft.CodeAnalysis.DocumentInfo.Create(Microsoft.CodeAnalysis.Document document) -> Microsoft.CodeAnalysis.DocumentInfo
```

Comment:
hrmm... would like @jasonmalinowski 's thoughts on this one.

Reply:
In principle I don't mind the API, but I'm not sure if we *really* need it. I'm not sure if we really need a fancy way of passing three strings to an API...? (especially when I'm not sure how to use all three...)

Reply:
Right now only the folders and filename are used. Really though, I was just surprised that creating `DocumentInfo` from a `Document` wasn't already part of the API. 

The benefit of using `DocumentInfo` is that it is an abstraction already of metadata about a document, which is what we want to change. Keeping that consistent has benefits for usability over just taking in two strings.